### ISSUE TITLE
No "Trusted TLS connection established to" message by default

### ISSUE THREAD

**Issue by kroeckx** [Snawoot/postfix-mta-sts-resolver#6]
As far as I know, most people run at the smtp_tls_security_level = may, and some at dane. As far as I know, that will never result in a "Trusted TLS connection established to" message. I get Untrusted and Anonymous connections, and Verified for the DANE enabled domains. Could you please update README.md to not suggest that people should already get Trusted connections?

**Comment by Snawoot** [comment]
Hello,

I use postfix 3.1.9 and it's source code has [following fragment](https://github.com/vdukhovni/postfix/blob/v3.1.9/postfix/src/tls/tls_client.c#L1146-L1153) for logging. Logging record depends solely on state of TLS session.

[Release notes](https://github.com/vdukhovni/postfix/blob/v3.1.9/postfix/RELEASE_NOTES-2.5#L336-L359) for earlier release, which introduced current state of TLS implementation, defines connection levels as follows:

> [Incompat 20080109] TLS logging output has changed to make it more
> useful. Existing logfile parser regular expressions may need
> adjustment.
> 
> - More log entries include the "hostnamename[ipaddress]" of the
>   remote SMTP peer.
> 
> - Certificate trust chain error reports show only the first
>   error certificate (closest to the trust chain root), and the
>   reporting is more human-readable for the most likely errors.
> 
> - After the completion of the TLS handshake, the session is logged
>   with TLS loglevel >= 1 as either "Untrusted", "Trusted" or
>   "Verified" (SMTP client only).
>   - "Untrusted" means that the certificate trust chain is invalid,
>     or that the root CA is not trusted.
>   - "Trusted" means that the certificate trust chain is valid, and
>     that the root CA is trusted.
>   - "Verified" means that the certificate meets the SMTP client's
>     matching criteria for the destination:
>     - In the case of a destination name match, "Verified" also
>       implies "Trusted".
>     - In the case of a fingerprint match, CA trust is not applicable.

In other words, `Anonymous` means absence of certificate (typical for incoming TLS connections), `Untrusted` means self-signed cert, `Trusted` means valid cert signed by public (or known) CA, `Verified` means valid cert with cert trust inferred from reliable source (CA, DANE or fingerprint) AND matching validation criteria.

By default with `smtp_tls_security_level = may` people should get `Trusted` for peers with valid cert, `Untrusted` for self-signed and always get `Anonymous` for incoming connections. After enabling MTA-STS people should get `Verified` against destinations with valid cert and published MTA-STS policy.

But newer Postfix versions possibly changed something and it worth checking. What is your Postfix version?

**Comment by kroeckx** [comment]
I'm running 3.3.2

**Comment by kroeckx** [comment]
Note that I'm currently not running mta-sts-resolver yet.
I also have smtp_tls_security_level = dnae

**Comment by kroeckx** [comment]
By default smtp_tls_CApath is empty it seems, changing that
changes things to a Trusted connection.

**Comment by Snawoot** [comment]
Thank you!

I have been missing that fact default value of `smtp_tls_CApath` and `smtp_tls_CAfile` is empty and no system certificates loaded. However, that hasn't been always true: some [discussions in google groups](https://groups.google.com/forum/#!msg/mailing.postfix.users/BruGNg4BuBU/1akyHTaA6d8J) and documentation for [tls_append_default_CA](http://www.postfix.org/postconf.5.html#tls_append_default_CA) directive indicate system certs was loaded by default prior to versions 2.4.15, 2.5.11, 2.6.8 and 2.7.2.

Current code ([version 3.3.2, for example](https://github.com/vdukhovni/postfix/blob/v3.3.2/postfix/src/tls/tls_certkey.c#L82-L115)) doesn't load any certificates at all if CApath or CAfile are empty, even if [tls_append_default_CA](http://www.postfix.org/postconf.5.html#tls_append_default_CA) is set to `yes`.

I'll have to somehow play around this aspect in documentation.

**Comment by Snawoot** [comment]
Resolved by de5f243
